core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/relational/InExpression.kt

```diff
@@ -15,3 +19,24 @@ class InExpression(
 
 fun <T : ValidType> TypeExpression<T>.inArray(array: TypeExpression<out ArrayType<out ValidType>>): InExpression =
     InExpression(this, array)
+
+fun Number.inArray(array: TypeExpression<out ArrayType<out ValidType>>): InExpression =
```

Comment:
I don't think we want this 'out'-statemets all over the place. It makes it less type safe. Here we want something like:
`fun Number.inArray(array: TypeExpression<ArrayType<NumberType>>): InExpression<NumberType> =`
Can you go over the whole project and remove these _out_ (maybe you have to add some more generic <T> on the way). I can only think of Arrays with different types where we might have to use an out.

Reply:
I deleted the ones i could fine and aren't necessary.